### ISSUE TITLE
Only-data dataset loaded unexpectedly as validation split

### ISSUE THREAD

**Issue by albertvillanova** [huggingface/datasets#3788]
## Describe the bug
As reported by @thomasw21 and @lhoestq, a dataset containing only a data file whose name matches the pattern `*dev*` will be returned as VALIDATION split, even if this is not the desired behavior, e.g. a file named `www_farodevigo_es.jsonl.gz`.

## Steps to reproduce the bug
```python
ds = load_dataset("bigscience-catalogue-lm-data/lm_es_pseudocrawl-filtered_373_www_farodevigo_es")
```

**Comment by lhoestq** [comment]
I see two options:
1. drop the "dev" keyword since it can be considered too generic
2. improve the pattern to something more reasonable, e.g. asking for a separator before and after "dev"
```python
["*[ ._-]dev[ ._-]*", "dev[ ._-]*"]
```

I think 2. is nice. If we agree on this one we can even decode to require the separation for the other split keywords "train", "test" etc.

**Comment by albertvillanova** [comment]
Yes, I had something like that on mind: "dev" not being part of a word.

**Comment by thomasw21** [comment]
Is there a reason why we want that regex? It feels like something that'll still be an issue for some weird case. "my_dataset_dev" doesn't match your regex, "my_dataset_validation" doesn't either ...

**Comment by lhoestq** [comment]
The regex is needed as part of our effort to make datasets configurable without code. In particular we define some dataset generic repository structures that users can follow

> ```
> "[^a-zA-Z]*dev[^a-zA-Z]*"
> ```

unfortunately our glob doesn't support "^": 

https://github.com/fsspec/filesystem_spec/blob/3e739db7e53f5b408319dcc9d11e92bc1f938902/fsspec/spec.py#L465-L479

**Comment by lhoestq** [comment]
> "my_dataset_dev" doesn't match your regex, "my_dataset_validation" doesn't either ... Why not always "train" unless specified?

And `my_dataset_dev.foo` would match the pattern, and we also have the same pattern but for the "validation" keyword so `my_dataset_validation.foo` would work too

**Comment by thomasw21** [comment]
> The regex is needed as part of our effort to make datasets configurable without code
This feels like coding with the filename ^^'

**Comment by lhoestq** [comment]
This is still much easier than having to write a full dataset script right ? :p
